### DISCUSSION TITLE
Announcement time (early - late)

### DISCUSSION THREAD

**Discussion by WolfgangZ42** [devemux86/cruiser#78] [Q&A] [answered]
In my settings speed for car is set to 30/60/160 and the announcement time to 25% .
Nevertheless I get the announcements at 1,9 km and 370 m, especially 1,9 km is too early for my opinion. How can I change that to ~1 km / ~300 m ?
My .pref file: [#T.zip](https://github.com/devemux86/cruiser/files/10947994/T.zip)

**Comment by devemux86** [comment]
Thanks for the .pref file, it always helps to better understand the questions / reports!

- Navigation profiles announcement describes how we can fully control the voice guidance: https://github.com/devemux86/cruiser/discussions/19

---

About your question:
- "Navigation instructions" controls the distances of voice prompts (on / off)
  - The distances are adjusted by the navigation profile's **average** speed
(min / max speeds are used in the new speed auto zoom: https://github.com/devemux86/cruiser/discussions/63)
- "Announcement time" controls the arrival announcements (earlier - later):
  - Arrive at turns: turn now without distance
  - Arrive at intermediate points or destination
- "Current speed" allows the current speed (GPS) to influence the voice guidance:
speak the voice prompts some seconds before the position has been reached

Better start with the default speeds and settings and check what happens on road or simulation?

For example, the default car / motorcycle average speed is 45 km / h,
that should lead to announcements at _approximately_ 300 m, 1.4 km, ...

---

It's difficult to have absolute fixed rounded distances with speed,
especially as we expect the speed to adjust the voice guidance.

However, _I am always available_ to discuss improvements or changes in Cruiser navigation! :slightly_smiling_face:

**Comment by devemux86** [comment]
Continuing the discussion from:

- https://github.com/devemux86/cruiser/discussions/63#discussioncomment-5293065

> Could you imagine to take the announcement time (early - late) for all announcements (not only for the last one)?

This will be available in the next Cruiser update.

Then we could keep the same influence of average speed in both functions?
- Voice prompts
- Route duration

**Reply by WolfgangZ42** [reply]
For me my 70 km/h average speed should calculate the route duration exactly and with let say 50 % announcement time adjustment the voice  should come at the perfect moment. Thanks a lot!

**Comment by devemux86** [comment, marked as answer]
> Could you imagine to take the announcement time (early - late) for all announcements (not only for the last one)?

Implemented in [Cruiser 3.0.13](https://play.google.com/store/apps/details?id=gr.talent.cruiser).

**Reply by WolfgangZ42** [reply]
Now I can play with the average speed definitions to get a realistic route duration. And adjust my announcement time (currently 75%) to avoid being woken up too early. 😄 
Thanks for making this possible!

**Reply by devemux86** [reply]
@WolfgangZ42 thanks for the requests and reports!

They help making Cruiser better and more user friendly for everyone.
If you need anything else, please let me know.

**Comment by 0709wiwiwi** [comment]
I am the only one ? 

If you often change types of vehicles (or just on foot), it remains unclear to me at what distance to expect a turncommanod. 
I find this setting a very complex way, too much hassle, or the  most unpredictable item in Cruiser. Too many factors come into play for (one) or more distance notifications ? At least I keep dabbling with it constantly, I still so don't fully understand it.  So the result surprises me every time and is actually not satisfactory.

Why not by a simple setting that is very clear though:  
"Belongs best with the menu where you set the avg speed per vehicle"

- Cycle bike announcements @ xxx, xx, x meters before turns.   
My setting: ---, -- , 50 meter before turns.
- Motorbike announcements @ xxx, xx, x meters before turns.   
Your setting: 800, 250, 120 meter before turns ? 
- etc.

**Reply by devemux86** [reply]
Voice prompts are triggered at specific times and with the navigation profile's (selected) average speed,
we can adjust the final distances we want the announcements to happen.

For example here are the _approximate_ times of the navigation voice prompts
and distances with the default average speed of 45 km/h for car / motorcycle:
- Large distance: 300s ~ 3750m
- Medium distance: 120s ~ 1500m
- Small distance: 24s ~ 300m
- Turn now: 4s ~ 50m

Different (selected) average speed leads to different distances.
Different profiles, with other speeds, lead to different distances.

"Announcement time" setting is a factor to influence all voice prompts to be announced earlier or later.

Voice system should be flexible to work with multiple navigation profiles (car, motorcycle, bike, walk)
and adjustable for everyone to adapt it to their driving style.

---

Or do you think that now that Cruiser has the new categorized (searchable) settings,
we should offer separate sliders for each navigation profile and each distance group?

**Comment by 0709wiwiwi** [comment]
No extra sliders ;-)
_The Motorbike 45kmh list (timing) along with "announcement time" factor shifting voice prompts earlier or later is a good clue to work on this further. To "tune" the important "small distance" announcement, it's best to do some maths anyway. Can the menu be improved so that the voice prompt distances are indicated ?_

Anyway I had some "timeconsuming" tests.
See the text attachment with my results.
[Announcement time_speed_distances.txt](https://github.com/devemux86/cruiser/files/11069257/Announcement.time_speed_distances.txt)

**Reply by devemux86** [reply]
@0709wiwiwi thanks very much for the tests and detailed info!

For example here are the _approximate_ times of the navigation voice prompts
and distances with the default average speed of 10 km/h for the bike profile:

- Large distance: 300s ~ 830m
- Medium distance: 120s ~ 330m
- Small distance: 24s ~ 65m
- Turn now: 4s ~ 12m

However, we should not need to understand how the algorithms of an app work.
We just want to choose settings and see immediate and understandable results.

I already plan to check how to offer more clear settings for voice prompt distances. :slightly_smiling_face:

**Reply by 0709wiwiwi** [reply]
I myself prefer the small distance announcement at about 15 seconds. Including some delay to generate TTS and without an additional delay prepare audio connection, in practice (18 kmh) equates to about 60 metres alert distance.  Fine. 
The influence of the 1 min avg speed is sure desirable, but important is that the lower threshold cannot get too low, as an announcement below 20 m is only correct in theory. 
In practice, you still have to deal with the gps vs chart tolerances which are more important especially at the slow speed announcements. 
You must make sure that a turn notification is thereby never too late.  A forest walk through summer leaves often results in quite deviating faulty GPS positions.

**Comment by devemux86** [comment]
_Cruiser has always been about the user's choices._ :slightly_smiling_face: 

I am finishing a new simpler implementation of the voice instructions:
- You will be able to control **exactly** where **all** voice instructions are announced
and also turn them on / off.

More details with images will be published soon in a separate topic.
